### ISSUE TITLE
Insights: Protect / Cameras

### ISSUE THREAD

**Issue by Waffull** [davidnewhall/unifi-poller#76]
I'm probably getting greedy here, but I'm a data geek. Does the CK2+/Protect and/or individual cameras, offer up stats the way the controller, switches, etc do?  I'd be happy to give you whatever logs you may need, if they would help.

Thanks.

**Comment by davidnewhall** [comment]
Start digging in the API :D Pretty much all I do to figure things out. I don't have any cameras/protect stuff, so not even sure where to start.  Do they use the same controller software?

**Comment by davidnewhall** [comment]
You can use the `-j "other /path/here"` cli arg to probe specific paths in the API. This will print out the raw reply. It's possibly slightly easier to use that setting up a curl cookiejar. You can see some of the paths I've put into the library [here](https://github.com/golift/unifi/blob/4ac11ccf94a5039bc180ca86ae8a50416fe4b117/types.go#L15-L28). There are other API (a big PHP one exists) that show more paths. The `%s` you see are usually replaced with the site name (`default` or the cryptic id). See if you can find (google) api paths for cameras. Start probing the data. Use `jq` to format and color the outpu json. Once you find data in an api path that you want to graph... post it! I need the API path (which will likely contain a site/`%s`) and an example json payload.

**Comment by Waffull** [comment]
Different controller software, called Protect https://unifi-protect.ui.com/
This may be beyond my Linux capabilities at this point.  I'm not even sure for what command the -j argument would be used.

I'll start looking to see what I can find.  Didn't even get to play around with the compiling info you sent yesterday... Was a busy day :(

**Comment by davidnewhall** [comment]
No worries. Yeah, different controller probably puts it out of scope for this project, at least for now. I like the UniFi camera software but I cannot fathom paying over $100 for a 2MP camera now days. All my 6MP Dahuas were ~$110, and there's newer nicer ones for not much more. Once UniFi has 4 MP PTZ with auto tracking at under $800 I'm on board.

As for the `-j` option this is something you can pass to `unifi-poller`. You can read a bit more about it in the [man page](https://github.com/davidnewhall/unifi-poller/blob/master/examples/MANUAL.md).
Using the `other` option allows you to request any url from your controller. You can do this with curl and a cookie jar too. That allows you to make an authenticated request, save the cookie and make more requests with that cookie. Using `-j` sorta takes care of that for you.

**Comment by Waffull** [comment]
So I found this in the community... Much of it is beyond my understanding
as I'm not a developer nor programmer.  I'm still all about writing DOS
batch files, lol.

https://community.ui.com/questions/Unifi-Protect-API-or-Motion-alerts-hit-API/b5c6cf44-651c-4751-98a8-c55d7ce9bd5f

I hear you about the expense of the cameras.  I'm just all about fully
integrated systems and am willing to pay extra for that convenience.  If I
was in my hay day of living on the bleeding edge of technology and working
in the field, I'd probably feel different.  But when it came time for me to
build my home network, after years of being out the field and in the board
room, I went with Unifi bc of it's simplicity of deployment and low price
point for near enterprise quality.

Anyway, going to find time to play around with all of this, but may not
happen until next week.

BTW, thank you for all your help and willingness to teach an old dog new
tricks ;)

**Comment by davidnewhall** [comment]
I have a fully integrated system. [SecuritySpy](http://www.bensoftware.com/securityspy/) handles the cameras, [Indigo](http://www.indigodomo.com/) handles the home automation and Unifi handles the network. They can all "speak" to each other one way or another and they all run on the same mac server presented by one nginx instance: https://github.com/autotyed/autotyed  I'm not sure yet what the advantage of having unifi cameras is if they're not technically integrated (you run another controller app). I'd pay extra too, but it's not about the camera price so much as the technology in the cameras is years and years old. I already wrote a 100% coverage [api library API consumer for securityspy](https://github.com/golift/securityspy) (equivalent to unifi protect). Seems to me full integration is not yet possible with unifi since their API is undocumented, it's just proprietary at this point.  One day, maybe in my next house ;)

**Comment by davidnewhall** [comment]
I'm sorta doubting Unfi will comment on API availability; they tend to shy away from such questions. Chances are, it has an API, you just need to find it.

**Comment by Waffull** [comment]
Wow, that api library is quite inclusive.  I've only purchased the G3 Pro cameras.  After installing all sorts of cameras over the years, I've been impressed with the quality of the G3 Pro and like the optical zoom.  For around my house, I don't really need pan & tilt and don't really like the way dome cameras look.  They feel like someone is trying to hide the camera.  I'll probably eventually get a few G4 Pro cameras.  I like that they have a slightly wider field of view.  I've never had a camera system at home before.  But I live in the middle of no where.  A half mile to the closest road and further to a house in which people live all year. I'm in a resort town, so a lot of weekend/holiday people.  I wanted to be able to know when someone comes up to the house, especially just curious people out on hiking/biking/ATV/Snowmobile trails, who decide to ignore the no trespassing signs.  But almost as important, I want to catch images/videos of wild life.  Just days before I installed the 1st of 3 cameras I have now, we saw mouse tracks.  Mouse are pretty rare around here.  But deer and bears can be seen often and when I eventually have all 11 cameras installed, I should get some pretty decent photos.  Especially after construction is finished and I no longer have excavators rumbling all day long, delivery trucks and of course my GC, his guys, the subs and myself.  Lol.  They also help me know when people are actually at the house working.  It's funny, I've been spending much of my days in the mechanical room working on networking, etc.  I can't see the front of the house or hear most of what's going on... So I keep the live feed on, even though I'm just downstairs.

Anyway, I'm totally rambling which is proof positive I'm over tired.  So I'll say goodnight.

**Comment by davidnewhall** [comment]
You totally need one of them 4 MP 30x auto tracking cams. You'll see wildlife like never before! That's pretty cool! 

Honestly, people ask me all the time about security systems and what they should get. I have a very custom system I've spent years putting together; it's not for everyone, but I have tried to give back and make some of my steps easy to follow.

I tell people they should get a system they can understand, operate and feel comfortable with. You've done exactly that! That's worth more than any other factors in my opinion. Many systems are complicated and things can certainly be a lot easier, manufacturers just need to pay attention to our pain points. I think Ubiquiti is on the right track for sure and I'm very happy to see other techs rocking their protect gear with praise. 

Thanks for the cool info, always refreshing!
